From eb8688c2b6cebb319511ca3102fc0f933adbafa2 Mon Sep 17 00:00:00 2001
From: panbingkun <panbingkun@baidu.com>
Date: Wed, 17 Apr 2024 23:09:06 -0700
Subject: [PATCH] [SPARK-47896][BUILD] Upgrade netty to `4.1.118.Final`

### What changes were proposed in this pull request?
The pr aims to upgrade `netty` from `4.1.108.Final` to `4.1.118.Final`.

### Why are the changes needed?
https://netty.io/news/2024/04/15/4-1-118-Final.html
This version has brought some bug fixes and improvements, such as:
- Fix DefaultChannelId#asLongText NPE ([#13971](https://github.com/netty/netty/pull/13971))
- Rewrite ZstdDecoder to remove the need of allocate a huge byte[] internally ([#13928](https://github.com/netty/netty/pull/13928))
- Don't send a RST frame when closing the stream in a write future while processing inbound frames ([#13973](https://github.com/netty/netty/pull/13973))

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

Closes #46112 from panbingkun/netty_for_spark4.

Authored-by: panbingkun <panbingkun@baidu.com>
Signed-off-by: Dongjoon Hyun <dhyun@apple.com>
---
 dev/deps/spark-deps-hadoop-3-hive-2.3 | 38 +++++++++++++--------------
 pom.xml                               |  2 +-
 2 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/dev/deps/spark-deps-hadoop-3-hive-2.3 b/dev/deps/spark-deps-hadoop-3-hive-2.3
index 54e54a1089041..71a87a9f519dd 100644
--- a/dev/deps/spark-deps-hadoop-3-hive-2.3
+++ b/dev/deps/spark-deps-hadoop-3-hive-2.3
@@ -198,16 +198,16 @@ metrics-jmx/4.2.25//metrics-jmx-4.2.25.jar
 metrics-json/4.2.25//metrics-json-4.2.25.jar
 metrics-jvm/4.2.25//metrics-jvm-4.2.25.jar
 minlog/1.3.0//minlog-1.3.0.jar
-netty-all/4.1.108.Final//netty-all-4.1.108.Final.jar
-netty-buffer/4.1.108.Final//netty-buffer-4.1.108.Final.jar
-netty-codec-http/4.1.108.Final//netty-codec-http-4.1.108.Final.jar
-netty-codec-http2/4.1.108.Final//netty-codec-http2-4.1.108.Final.jar
-netty-codec-socks/4.1.108.Final//netty-codec-socks-4.1.108.Final.jar
-netty-codec/4.1.108.Final//netty-codec-4.1.108.Final.jar
-netty-common/4.1.108.Final//netty-common-4.1.108.Final.jar
-netty-handler-proxy/4.1.108.Final//netty-handler-proxy-4.1.108.Final.jar
-netty-handler/4.1.108.Final//netty-handler-4.1.108.Final.jar
-netty-resolver/4.1.108.Final//netty-resolver-4.1.108.Final.jar
+netty-all/4.1.118.Final//netty-all-4.1.118.Final.jar
+netty-buffer/4.1.118.Final//netty-buffer-4.1.118.Final.jar
+netty-codec-http/4.1.118.Final//netty-codec-http-4.1.118.Final.jar
+netty-codec-http2/4.1.118.Final//netty-codec-http2-4.1.118.Final.jar
+netty-codec-socks/4.1.118.Final//netty-codec-socks-4.1.118.Final.jar
+netty-codec/4.1.118.Final//netty-codec-4.1.118.Final.jar
+netty-common/4.1.118.Final//netty-common-4.1.118.Final.jar
+netty-handler-proxy/4.1.118.Final//netty-handler-proxy-4.1.118.Final.jar
+netty-handler/4.1.118.Final//netty-handler-4.1.118.Final.jar
+netty-resolver/4.1.118.Final//netty-resolver-4.1.118.Final.jar
 netty-tcnative-boringssl-static/2.0.61.Final//netty-tcnative-boringssl-static-2.0.61.Final.jar
 netty-tcnative-boringssl-static/2.0.65.Final/linux-aarch_64/netty-tcnative-boringssl-static-2.0.65.Final-linux-aarch_64.jar
 netty-tcnative-boringssl-static/2.0.65.Final/linux-x86_64/netty-tcnative-boringssl-static-2.0.65.Final-linux-x86_64.jar
@@ -215,15 +215,15 @@ netty-tcnative-boringssl-static/2.0.65.Final/osx-aarch_64/netty-tcnative-borings
 netty-tcnative-boringssl-static/2.0.65.Final/osx-x86_64/netty-tcnative-boringssl-static-2.0.65.Final-osx-x86_64.jar
 netty-tcnative-boringssl-static/2.0.65.Final/windows-x86_64/netty-tcnative-boringssl-static-2.0.65.Final-windows-x86_64.jar
 netty-tcnative-classes/2.0.65.Final//netty-tcnative-classes-2.0.65.Final.jar
-netty-transport-classes-epoll/4.1.108.Final//netty-transport-classes-epoll-4.1.108.Final.jar
-netty-transport-classes-kqueue/4.1.108.Final//netty-transport-classes-kqueue-4.1.108.Final.jar
-netty-transport-native-epoll/4.1.108.Final/linux-aarch_64/netty-transport-native-epoll-4.1.108.Final-linux-aarch_64.jar
-netty-transport-native-epoll/4.1.108.Final/linux-riscv64/netty-transport-native-epoll-4.1.108.Final-linux-riscv64.jar
-netty-transport-native-epoll/4.1.108.Final/linux-x86_64/netty-transport-native-epoll-4.1.108.Final-linux-x86_64.jar
-netty-transport-native-kqueue/4.1.108.Final/osx-aarch_64/netty-transport-native-kqueue-4.1.108.Final-osx-aarch_64.jar
-netty-transport-native-kqueue/4.1.108.Final/osx-x86_64/netty-transport-native-kqueue-4.1.108.Final-osx-x86_64.jar
-netty-transport-native-unix-common/4.1.108.Final//netty-transport-native-unix-common-4.1.108.Final.jar
-netty-transport/4.1.108.Final//netty-transport-4.1.108.Final.jar
+netty-transport-classes-epoll/4.1.118.Final//netty-transport-classes-epoll-4.1.118.Final.jar
+netty-transport-classes-kqueue/4.1.118.Final//netty-transport-classes-kqueue-4.1.118.Final.jar
+netty-transport-native-epoll/4.1.118.Final/linux-aarch_64/netty-transport-native-epoll-4.1.118.Final-linux-aarch_64.jar
+netty-transport-native-epoll/4.1.118.Final/linux-riscv64/netty-transport-native-epoll-4.1.118.Final-linux-riscv64.jar
+netty-transport-native-epoll/4.1.118.Final/linux-x86_64/netty-transport-native-epoll-4.1.118.Final-linux-x86_64.jar
+netty-transport-native-kqueue/4.1.118.Final/osx-aarch_64/netty-transport-native-kqueue-4.1.118.Final-osx-aarch_64.jar
+netty-transport-native-kqueue/4.1.118.Final/osx-x86_64/netty-transport-native-kqueue-4.1.118.Final-osx-x86_64.jar
+netty-transport-native-unix-common/4.1.118.Final//netty-transport-native-unix-common-4.1.118.Final.jar
+netty-transport/4.1.118.Final//netty-transport-4.1.118.Final.jar
 objenesis/3.3//objenesis-3.3.jar
 okhttp/3.12.12//okhttp-3.12.12.jar
 okio/1.15.0//okio-1.15.0.jar
diff --git a/pom.xml b/pom.xml
index 7ded74b9f9dfc..e6b37610adb23 100644
--- a/pom.xml
+++ b/pom.xml
@@ -217,7 +217,7 @@
     <bouncycastle.version>1.78</bouncycastle.version>
     <tink.version>1.13.0</tink.version>
     <datasketches.version>5.0.1</datasketches.version>
-    <netty.version>4.1.108.Final</netty.version>
+    <netty.version>4.1.118.Final</netty.version>
     <netty-tcnative.version>2.0.65.Final</netty-tcnative.version>
     <icu4j.version>72.1</icu4j.version>
     <!--
